### PR TITLE
qa/spelling: Add Codespell support

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

This PR adds support for [Codespell](https://github.com/codespell-project/codespell) to detect common misspellings and fixes any flagged issues.

As is the nature with any such system, sometimes it can detect false positives. As such, words and lines can be added to the newly created files in [./contrib/spellcheck](https://github.com/thrasher-/gocryptotrader/tree/codespell/contrib/spellcheck) as there is no inline support yet.

Currently the GitHub action doesn't support specifying a config so the `skip` variable needs to be updated whenever there is a change made on the `.codespellrc` file

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] Fancy new GH action

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/spell-checker.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+jobs:
+  codespell:
+    name: Spell checker
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: codespell-project/actions-codespell@master
+        with:
+          exclude_file: contrib/spellcheck/exclude_lines.txt
+          skip: ./.git,./web,./currency/code_types.go,*.json,*.sum
```
Comment (gloriousCode) [inline .github/workflows/spell-checker.yml:12]: Can we add *.html? Or we can also exclude `./cmd/exchange_wrapper_issues/report.html` as that's being flagged even though its gitignored

File: database/repository/trade/trade_test.go [hunk 2]
```diff
@@ -209,7 +209,7 @@ func tradeSQLTester(t *testing.T) {
 		t.Error(err)
 	}
 	if len(v) != 0 {
-		t.Errorf("should all be ded %v", v)
+		t.Errorf("should all be dead %v", v)
```
Comment (gloriousCode) [inline database/repository/trade/trade_test.go:212]: :laughing:

**Review by gloriousCode — CHANGES_REQUESTED** [review]
tthis iz a vry xcitng chanje1!

**Review by gloriousCode — APPROVED** [review]
I rescind my previous comment because re-rerunning exchange_wrapper_issues with all exchanges has resolved things (even though its not part of the changes :shrug: )

tACK!

File: exchanges/kraken/futures_types.go [hunk 3]
```diff
@@ -401,7 +401,7 @@ type FuturesNotificationData struct {
 		Priority         string `json:"priority"`
 		Note             string `json:"note"`
 		EffectiveTime    string `json:"effectiveTime"`
-	} `json:"notifcations"`
+	} `json:"notifications"`
```
Comment (shazbert) [inline exchanges/kraken/futures_types.go:404]: 🦅 I dont' have futures api keys but this matches docs.

**Review by shazbert — APPROVED** [review]
LGTM tACK!

**Pull request merged**